### PR TITLE
Ensure postprocessors are computed during linear solve with PJFNKMO

### LINKED ISSUE #28993: Need to make Problem/bx_norm work with PJFNKMO eigenvalue calculation
### Motivation

When adding `Problem/bx_norm` parameter with `Executioner/solve_type=PJFNKMO` with `Eigenvalue` executioner, calculations will produce wrong results.

### Design

Users would expect `Problem/bx_norm` works regardless the solve_type. This turns out to be passing the unnecessary normalization to SLEPc in case of PJFNKMO. A test will be added after the code fix.

### Impact

A combination of input parameters that possibly has not be used by users so far.

### PR DESCRIPTION
**Pull request by lindsayad**
Closes #28993

### CODE REVIEW

File: test/tests/problems/eigen_problem/jfnk_mo/tests [hunk 1]
```diff
@@ -57,6 +57,7 @@
     type = 'CSVDiff'
     input = 'ne_coupled_mo.i'
     csvdiff = 'ne_coupled_eigenvalues_0001.csv'
+    cli_args = 'Problem/bx_norm=uint'
```
Comment (YaqiWang) [inline test/tests/problems/eigen_problem/jfnk_mo/tests:60]: I feel this deserve a separate test with its own requirement.

File: framework/src/utils/SlepcSupport.C [hunk 2]
```diff
@@ -951,6 +951,13 @@ mooseSlepcEigenFormFunctionB(SNES snes, Vec x, Vec r, void * ctx)
 
     ierr = MatMult(B, x, r);
     CHKERRQ(ierr);
+
+    if (eigen_problem->bxNormProvided())
+    {
+      // User has provided a postprocessor. We need it updated
+      updateCurrentLocalSolution(eigen_nl.sys(), x);
+      eigen_problem->execute(EXEC_LINEAR);
```
Comment (YaqiWang) [inline framework/src/utils/SlepcSupport.C:959]: I realized that we do not have a check to make sure `execute_on` of the bx_norm postprocessor includes linear. Care to add it in `EigenProblem`?

**Review by YaqiWang — CHANGES_REQUESTED** [review]
The code change looks good. I checked with a Griffin test and it does work as expected! Just need to address my two comments and this should be good to go.

**Comment by lindsayad** [comment]
Nice suggestions. Hopefully should be addressed

**Review by GiudGiud — APPROVED** [review]

**Comment by YaqiWang** [comment]
My comments are addressed. I cannot resolve the comments by myself.

**Comment by lindsayad** [comment]
thanks for raising the issue and the review @YaqiWang !

**Pull request merged**